examples/step-31/step-31.cc

```diff
@@ -1040,15 +1054,17 @@ namespace Step31
     // and $\mathbf u^{n-2}$, as well as for the temperatures $T^{n}$,
     // $T^{n-1}$ and $T^{n-2}$ (required for time stepping) and all the system
     // right hand sides to their correct sizes and block structure:
-    stokes_solution.reinit (stokes_block_sizes);
-    old_stokes_solution.reinit (stokes_block_sizes);
-    stokes_rhs.reinit (stokes_block_sizes);
+    IndexSet temperature_partitioning (n_T);
+    temperature_partitioning.add_range(0,n_T);
+    stokes_solution.reinit (stokes_block_sizes, MPI_COMM_WORLD);
```

Comment:
Ha, who knew? :-) Someone must have had good foresight! In any case, @Rombur , I think this will make many of your changes much simpler because you can continue to store the size of a block and just create the index set on the fly.